### ISSUE TITLE
Conflict with dispatch:mocha-phantomjs?

### ISSUE THREAD

**Issue by gadicc** [practicalmeteor/meteor-mocha#23]
I'm running like this:

```
$ meteor test --driver-package practicalmeteor:mocha
```

And everything works.  But if I add `dispatch:mocha-phantomjs` to my `.meteor/packages`, with no other changes, server tests no longer run, I get:

```
passes: 0  failures: 0  duration: 0s
```

In case this doesn't affect all cases, there's a repro at https://github.com/gadicc/meteor-blaze-react-component/commit/6a0e529fc888f5c5485d3df17699d76d5855b860.

**Comment by rbabayoff** [comment]
@gadicc , yes, we don't co-exist nicely, since both of us are adding mocha independently. I suggest you wait a couple of days until we get support for 1.3 in spacejam and practicalmeteor:mocha-console-runner. 

Long term we hope to create a pr into mocha with the meteor specific support of having multiple html reporters to eliminate those things altogether, so you could just npm install mocha independently of any of the two packages.

**Comment by aldeed** [comment]
@rbabayoff I now have all the `dispatch:mocha-*` packages depending on [dispatch:mocha-core](https://github.com/DispatchMe/meteor-mocha-core), which loads the client side file and exports a single instance of mocha on the server for them all to share. This allows them all to be added without conflicts. I haven't look into it yet, but I believe practicalmeteor:mocha could also be updated to depend on dispatch:mocha-core and then everything would not conflict. I can give you access to dispatch:mocha-core. There isn't much there except that it has the mocha npm dependency, so we would have to bump that version occasionally.

**Comment by tmeasday** [comment]
@rbabayoff what do you think about trying to settle on a single mocha core package?

**Comment by rbabayoff** [comment]
@tmeasday I agree, but I'd prefer to use ours. PRs are welcome, 2 options:
1. Create the mocha instance in our mocha-core and add dependency on npm.
2. Have dispatch depend on practicalmeteor:mocha and use the exported mocha instance from our package. Preferred.

@aldeed and @tmeasday what say you?

**Comment by tmeasday** [comment]
I'm really not sure what the difference between the core packages are? Why would you prefer to use yours @rbabayoff?

**Comment by aldeed** [comment]
@rbabayoff Our core pkg already depends on your `mocha-core` to do the binding, so if your `mocha-core` just exported the mocha stuff already bound on the server and did init mocha on the client, that's my preference. In other words, we can merge ours into yours. I can do a PR but probably not until next week.

**Comment by aldeed** [comment]
@rbabayoff @tmeasday Here is a PR for combining the pkgs: https://github.com/practicalmeteor/meteor-mocha-core/pull/6

**Comment by jsep** [comment]
Hello @tmeasday @aldeed @gadicc 

 I just release a release candidate version (2.4.5-rc3.1) of the package using `practicalmeteor:mocha-core` with the changes of @aldeed 

To use it just run  `meteor add practicalmeteor:mocha@2.4.5-rc3.1`

Please feel free to test the changes and let us know any problem.

Thanks

**Comment by jsep** [comment]
@aldeed which version of your package it's using the same `practicalmeteor:mocha-core` package?

**Comment by aldeed** [comment]
Our packages `dispatch:mocha`, `dispatch:mocha-phantomjs`, and `dispatch:mocha-browser` all use our `dispatch:mocha-core`, which uses `practicalmeteor:mocha-core` to do the server wrapping. I will update those three packages to use `practicalmeteor:mocha@2.4.5-rc3.1` directly, and then `dispatch:mocha-core` will be deprecated. I will try to get our own RCs out for that so that we can verify that everything plays nicely together.

**Comment by jsep** [comment]
@aldeed the version of `practicalmeteor:mocha-core` that I'm using  is `1.0.0-rc.2`.

**Comment by Ben305** [comment]
@aldeed Can you post here again when the RCs are out?

**Comment by rbabayoff** [comment]
@gadicc, @aldeed and @tmeasday, as soon as you confirm it's working, we'll publish this as a release version.

**Comment by abhiaiyer91** [comment]
When is this being resolved? @tmeasday

**Comment by tmeasday** [comment]
Just waiting to hear from @aldeed . 

You can just use the dispatch reporter in the meantime if it's slowing you down @abhiaiyer91

**Comment by gadicc** [comment]
Sorry for my late replies to this... I'm a bit behind and also have some weird new issue with my testing setup that I've run out of time to fix.  In any event, I don't think I can test this without an updated `dispatch:mocha-phantomjs`, and I think as soon as they give the final OK you can consider this resolved.  Many thanks!  Will be awesome to have both `meteor test` and CI working with the same config.

**Comment by M4v3R** [comment]
@gadicc If this is fixed you may want to update [the guide](http://guide.meteor.com/testing.html#mocha) to remove this line:

> NOTE: Currently tests may not run properly if you add both a practicalmeteor:* package and a dispatch:* package to your app. This is being fixed. See this issue.

**Comment by maciej-trebacz** [comment]
@tmeasday  If this is fixed you may want to update [the guide](http://guide.meteor.com/testing.html#mocha) to remove this line:

> NOTE: Currently tests may not run properly if you add both a practicalmeteor:\* package and a dispatch:\* package to your app. This is being fixed. See this issue.

**Comment by rbabayoff** [comment]
@aldeed let us know if this works for you and if not, we'll reopen this.

**Comment by aldeed** [comment]
@rbabayoff @jsep I published new releases of all the `dispatch:mocha*` packages using `practicalmeteor:mocha-core@1.0.0`. As far as I can tell, everything is working well and there are no longer conflicts between the packages.

@tmeasday Here is a PR updating todos app: https://github.com/meteor/todos/pull/153 And the note in the guide should be updated as mentioned by @M4v3R above. If the latest release of all mocha packages is used, there is no conflict.

Thanks all!

**Comment by jsep** [comment]
Those are great news, thank you so much @aldeed !
